File: alzir/Entregável #1 - Algoritmos/JavaScript/contagem.js
const readline = require("readline").createInterface({
  input: process.stdin,
  output: process.stdout,
});

let numeros = [];

const filtrar = (numeros) => {
  const inteiros = numeros.filter((item) => item % 1 === 0 && item > 0);
  return console.log(inteiros.length, inteiros);
};

const perguntar = (quantidade) => {
  readline.question(
    `Digite o número (${numeros.length + 1} de ${quantidade}): `,
    (numero) => {
      numeros.push(Number(numero));

      if (quantidade <= numeros.length) {
        readline.close();
        filtrar(numeros);
      } else {
        perguntar(quantidade);
      }
    }
  );
};

readline.question("Digite a quantidade de números: ", (quantidade) => {
  if (isNaN(quantidade) || quantidade <= 0) {
    console.log("Digite um número inteiro positivo!");
    readline.close();
  } else perguntar(quantidade.trim());
});

File: alzir/Entregável #1 - Algoritmos/Java/src/Contagem.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class Contagem {

  static List<String> numeros = new ArrayList<>();

  public static List<String> filtrar(List<String> numeros) {
    List<String> inteiros = new ArrayList<>();
    for (String item : numeros) {
      int numero = Integer.parseInt(item);
      if (numero > 0) {
        inteiros.add(item);
      }
    }
    System.out.println(inteiros.size() + " " + inteiros);
    return inteiros;
  }

  public static void perguntar(int quantidade) {
    Scanner sc = new Scanner(System.in);
    System.out.print(
      "Digite o número (" + (numeros.size() + 1) + " de " + quantidade + "): "
    );
    String numero = sc.nextLine();
    numeros.add(numero);

    if (quantidade <= numeros.size()) {
      sc.close();
      filtrar(numeros);
    } else {
      perguntar(quantidade);
    }
  }

  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    System.out.print("Digite a quantidade de números: ");
    String quantidadeString = sc.nextLine();
    try {
      int quantidade = Integer.parseInt(quantidadeString);
      perguntar(quantidade);
    } catch (NumberFormatException e) {
      System.out.println("Você não digitou um número!");
    }
    sc.close();
  }
}
